frontend/public/components/utils/webhooks.tsx

```diff
@@ -90,27 +131,36 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
     }
 
     const secretName = _.get(trigger, [triggerProperty, 'secretReference', 'name']);
-    k8sGet(SecretModel, secretName, namespace).then(
-      (secret: K8sResourceKind) => {
-        if (!_.has(secret, 'data.WebHookSecretKey')) {
-          errorModal({
-            error: `Secret referenced in the ${triggerProperty} webhook trigger does not contain 'WebHookSecretKey' key. Webhook trigger won’t work due to the invalid secret reference`,
-          });
-          return;
-        }
-        const webhookSecretValue = Base64.decode(secret.data.WebHookSecretKey);
-        const url = getWebhookURL(trigger, webhookSecretValue);
-        navigator.clipboard.writeText(url);
-      },
-      (err) => {
-        const error = err.message;
-        errorModal({ error });
-      },
+    const webhookSecret: K8sResourceKind = webhookSecrets.find(
+      (secret: K8sResourceKind) => secret.metadata.name === secretName,
     );
+    if (!webhookSecret) {
+      errorModal({
+        error: `Could not get ${secretName} secret`,
```

Comment:
Instead of showing an error, I would just remove the copy link when the secret is missing. We know it won't work.

Reply:
We might want to also show the secret name as plain text instead of a link if an error occurred fetching it. Most likely it doesn't exist and will be a broken link. Checking for a 404 status code would tell us for sure.